real_time/surface_src/java_src/Alice/src/main/java/verifier/Verifier.java

```diff
@@ -0,0 +1,88 @@
+package verifier;
+import java.io.*;
+import java.util.*;
+
+import com.roboclub.robobuggy.messages.ImuMeasurement;
+import com.roboclub.robobuggy.ros.Message;
+import com.roboclub.robobuggy.ros.MessageListener;
+import com.roboclub.robobuggy.ros.Publisher;
+import com.roboclub.robobuggy.ros.SensorChannel;
+import com.roboclub.robobuggy.ros.Subscriber;
+
+//TODO: Write a VerifierChannel (like SensorChannel in BuggyRos), then make an ImuVerifierNode like ImuNode and move this code there.
+
+public class Verifier {
+	//Dim global variables:
+	static double x, y, z, a=0, b=0, c=0;
+	static boolean started = false;
+	static int count = 0;
+	static double[] thresh = {5, 8, 10}; //Thresholds determined empirically from moving files (roughly 1.5*maxChange)
+
+
+	//public static void main(String[] args) throws FileNotFoundException {
+	public Verifier(){	
+		//Potentially real-time code
+		Publisher p = new Publisher("IMUstatus");
+		Subscriber output = new Subscriber(SensorChannel.IMU.getMsgPath(), new MessageListener(){
+			@Override
+				public void actionPerformed(String topicName, Message m) {
+			ImuMeasurement mydata = (ImuMeasurement) m;
+			checkData(mydata, p);
+		}
+		});
+		
+		//Testing code; reads from a file
+		/*Scanner scanner = new Scanner(new File("/Users/davidneiman/Desktop/CMU/Roboclub/sensorsMoving.txt"));
+		while(scanner.hasNext()){
+			checkData(scanner.nextLine());
+		}
+		scanner.close();*/
+	}
+	
+	//For offline code
+	public static void checkData(String inp){
+		StringTokenizer st = new StringTokenizer(inp, ",");
+		st.nextToken();
+		String w = st.nextToken();
+		String s = st.nextToken();
+		if(!s.equals("START")){ //Check for non-data lines
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1334102760/issues/source?bid=2938665&fileBranchId=2957575#l48)